cedar-policy/src/api.rs

```diff
@@ -1701,9 +1701,10 @@ impl<'a> Diagnostic for ValidationWarning<'a> {
 
 /// Identifier portion of the [`EntityUid`] type.
 ///
-/// An `EntityId` can can be constructed using [`EntityId::from_str`] or by
-/// calling `parse()` on a string. This implementation is `Infallible`, so the
-/// parsed `EntityId` can be extracted safely.
+/// All strings are valid [`EntityId`]s, and can be
+/// constructed either using [`EntityId::new`]
+/// or by using the implementation of [`FromStr`]. This implementation is [`Infallible`], so the
+/// parsed [`EntityId`] can be extracted safely.
```

Comment:
Should probably update the example to use `new`

Reply:
Sure, I want to leave the `.parse()` example because it's slightly weird to newcomers

Reply:
Move the `parse` example to the `FromStr` doc string? Most users should be using `new` to construct this